File: src/Gateway/Domovoy.DbGateway/Models/Device.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Domovoy.DbGateway.Models;

public class Device
{
    [BsonId]
    public string DeviceId { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string Type { get; set; } = null!;

    public string LocationId { get; set; } = null!;

    public string Status { get; set; } = "Offline";

    [BsonElement("Config")]
    public Dictionary<string, object> Configuration { get; set; } = new();

    public DateTime LastSeen { get; set; } = DateTime.UtcNow;

    public bool IsOnline { get; set; }
}

File: src/UI/WebUI/src/types/device.ts
// Device Types for WebUI
// Validates: Requirements 1.1, 2.1

export type DeviceType = 'Light' | 'Sensor' | 'Switch' | 'Thermostat' | 'Unknown';

export type DeviceStatus = 'Online' | 'Offline' | 'Error';

export interface Device {
  deviceId: string;
  name: string;
  type: DeviceType;
  locationId: string;
  status: DeviceStatus;
  isOnline: boolean;
  lastSeen: Date;
  configuration: Record<string, any>;
}

export interface Light {
  lightId: string;
  deviceId: string;
  dimmable: boolean;
  colorSupport: boolean;
  defaultBrightness: number;
}

export interface DeviceCommand {
  deviceId: string;
  command: string;
  parameters: Record<string, any>;
}

export interface LightCommand extends DeviceCommand {
  command: 'toggle' | 'setBrightness' | 'setColor';
  parameters: {
    state?: boolean;
    brightness?: number;
    color?: string;
  };
}
